core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -806,15 +808,20 @@ private[spark] object Utils extends Logging {
         // It's more proper to pick ip address following system output order.
         val activeNetworkIFs = NetworkInterface.getNetworkInterfaces.toList
         val reOrderedNetworkIFs = if (isWindows) activeNetworkIFs else activeNetworkIFs.reverse
+
         for (ni <- reOrderedNetworkIFs) {
-          for (addr <- ni.getInetAddresses if !addr.isLinkLocalAddress &&
-               !addr.isLoopbackAddress && addr.isInstanceOf[Inet4Address]) {
+          val addresses = ni.getInetAddresses.toList
+            .filter { addr => !addr.isLinkLocalAddress && !addr.isLoopbackAddress }
+          if ( !addresses.isEmpty ) {
+            val addr = addresses.find( _.isInstanceOf[Inet4Address] ).getOrElse(addresses.head)
+            // because of Inet6Address.toHostName may add interface at the end if it knows about it
+            val strippedAddress = InetAddress.getByAddress(addr.getAddress)
```

Comment:
Ok, i agree with that. Should i make additional commit or change existing one?
*```nonEmpty```